hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/TargetManagement.java

```diff
@@ -251,23 +219,23 @@ long countByFilters(Collection<TargetUpdateStatus> status, Boolean overdueState,
      *             if distribution set with given ID does not exist
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_TARGET)
-    long countByRsqlAndNonDS(long distributionSetId, @NotNull String rsqlParam);
+    long countByNotAssignedDistributionSetAndQuery(long distributionSetId, @NotNull String rsqlParam);
 
     /**
      * Finds all targets for all the given parameter {@link TargetFilterQuery}
      * and that are not assigned to one of the {@link RolloutGroup}s
      *
-     * @param pageRequest
-     *            the pageRequest to enhance the query for paging and sorting
+     * @param limit
+     *            the maximum amount of results requested (could be less)
      * @param groups
      *            the list of {@link RolloutGroup}s
      * @param rsqlParam
      *            filter definition in RSQL syntax
      * @return a page of the found {@link Target}s
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_READ_TARGET)
-    Page<Target> findByTargetFilterQueryAndNotInRolloutGroups(@NotNull Pageable pageRequest,
-            @NotEmpty Collection<Long> groups, @NotNull String rsqlParam);
+    List<Target> findByQueryAndNotInRolloutGroups(int limit, @NotEmpty Collection<Long> groups,
```

Comment:
Returning a page would mean that I need to know the total-size (all targets not in a certain rollout-group). 
- Calculation of this count is very expensive, we always have to fetch all targets matching the filter.
- This method is only used within the rollout-start logic (maybe makes no sense to have it in the interface?) and I noticed, that the total-count within the page was never used. The group is filled and all these targets drop-out of this select because the get a group assigned.

Reply:
I see, than just go for a `Slice` instead of a `Page`:

```
Slice<Target> findByTargetFilterQueryAndNotInRolloutGroups(Pageable pageRequest, ...)
```

Reply:
Use of slice is sligthly better than page - still it is for no use (out of the sight of the current logic) and makes the implementation here more complicate. 
- We need to fetch all results until we have found the matching start of the slice
- We have to fill up the slice
- We have to fetch all results as until we are sure, if there is a next or there is no next slice

I understand the point from a interface-design point of view. still I would prefere to get rid of this method completely and find another way in the rollout-start logic.